### PR TITLE
feat: implement nonfugible provider by zerion

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Implement FT/NFT/Coin/Gas Provider by `Zerion`
Closes #MF-2350

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-providers/src/zerion/type.ts [hunk 1]
```diff
@@ -112,6 +117,59 @@ export interface ZerionTransactionItem {
...
+            type: string
+        }
+    }
+    name: string
+    type: string
+    tags: any
+    last_price: any
+    buy_now_price: any
+    floor_price: number
+    total_last_price: any
```
Comment (guanbinrui) [inline packages/web3-providers/src/zerion/type.ts]: Please avoid using `any`.

File: packages/web3-providers/src/zerion/index.ts [hunk 2]
```diff
@@ -170,3 +216,72 @@ export class ZerionAPI
         return createPageable(transactions, createIndicator(indicator))
     }
 }
+
+export class ZerionNonFungibleTokenAPI implements NonFungibleTokenAPI.Provider<ChainId, SchemaType> {
+    createNonFungibleTokenPermalink(address?: string, tokenId?: string) {
+        if (!address || !tokenId) return
+        return ZERION_NFT_DETAIL_URL + `${address}:${tokenId}`
```
Comment (guanbinrui) [inline packages/web3-providers/src/zerion/index.ts:102]: The NFT details link could vary for different EVM chains.

File: packages/web3-providers/src/zerion/index.ts [hunk 3]
```diff
@@ -170,3 +216,72 @@ export class ZerionAPI
...
+                chainId,
+                schema: nft.standard === 'ERC1155' ? SchemaType.ERC1155 : SchemaType.ERC721,
+                address: nft.asset.contract_address,
+                name: nft.asset.collection.name,
+                symbol: nft.asset.symbol,
+            },
+            metadata: {
+                chainId,
+                name: nft.asset.collection.name,
+                symbol: nft.asset.symbol,
```
Comment (guanbinrui) [inline packages/web3-providers/src/zerion/index.ts:122]: Doesn't the API response with an `imageURL` or a `mediaURL`?

File: packages/web3-providers/src/zerion/index.ts [hunk 4]
```diff
@@ -170,3 +216,72 @@ export class ZerionAPI
...
+            contract: {
+                chainId,
+                schema: nft.standard === 'ERC1155' ? SchemaType.ERC1155 : SchemaType.ERC721,
+                address: nft.asset.contract_address,
+                name: nft.asset.collection.name,
+                symbol: nft.asset.symbol,
+            },
+            metadata: {
+                chainId,
+                name: nft.asset.collection.name,
```
Comment (guanbinrui) [inline packages/web3-providers/src/zerion/index.ts]: It's not a collection name. Any other options?

File: packages/web3-providers/src/zerion/index.ts (hunk 2 above)
Comment (nuanyang233) [reply packages/web3-providers/src/zerion/index.ts:102]: Here is just the link to zerion's detail page. And zerion currently only supports Ethereum

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**